pandas/core/reshape/concat.py

```diff
@@ -496,7 +496,19 @@ def get_result(self):
                     # 1-ax to convert BlockManager axis to DataFrame axis
                     obj_labels = obj.axes[1 - ax]
                     if not new_labels.equals(obj_labels):
-                        indexers[ax] = obj_labels.reindex(new_labels)[1]
+                        # We have to remove the duplicates from obj_labels
```

Comment:
can't you do
`obj_labels = obj_labels.unique()` ? pandas keeps these in the same order

Reply:
No this does not work here unfortunately. 

The new_labels array may contain multiple duplicates. For example something like [1, 2, 2, 3, 3]
If obj_labels = [1, 2, 2, 3] we can only delete the duplicates which are also duplicated in obj_labels. The duplicates from the other inputs can not be deleted. That is the reason I did it this way.

Reply:
Sorry that did not answer your question. The reason we can not do ``obj_labels = obj_labels.unique()`` is different.

If new_labels = [1, 2, 2, 3, 3] and obj_labels = [1, 2, 2, 3]. if we would drop the duplicates, we would select the element at position 1 twice and the element at position 2 also twice, but for index 3. Also we would not select the element associated with index 3. We can not change the lenght of obj_labels without side effects, we have to modify new_labels

Reply:
Here is an example with obj_labels.unique()

```
    df1 = pd.DataFrame([1, 2, 3, 4], index=[0, 1, 1, 4], columns=["a"])
    df2 = pd.DataFrame([6, 7, 8, 9], index=[0, 0, 1, 3], columns=["b"])
    result = pd.concat([df1, df2], axis=1)
```
result in
```
    a    b
0  1.0  6.0
0  1.0  6.0
1  2.0  7.0
1  2.0  7.0
3  NaN  8.0
4  3.0  NaN
```

but clearly should equal
```
     a    b
0  1.0  6.0
0  1.0  7.0
1  2.0  8.0
1  3.0  8.0
3  NaN  9.0
4  4.0  NaN
```

Reply:
this is a highly sensitive part of the code so

a) make a function to do this. I *really* think you are duplicating a lot of existing code, but it could go in core/algorithms.py. I *know* we do this in places in core/indexing.py as well.
b) benchmark this.

Reply:
Could you give me a pointer as to where to look in indexing? I checked all functions in there but could not find a part where we do something similar like what we have to do here.